php/libraries/UserPermissions.class.inc

```diff
@@ -106,30 +106,83 @@ class UserPermissions
 
 
     /**
-     * Determines if the user has a permission
+     * Determines if the user has the permission to access a specific module
+     * or page.
      *
      * @param string $code The permission code
      *
-     * @return bool
+     * @return bool If the user has a permission.
      */
-    function hasPermission($code)
+    function hasPermission(string $code): bool
```

Comment:
since this is going to major and we're changing the type signature, maybe we should take the opportunity to explicitly declare this public too.